htdocs/xoops_lib/Xoops/Core/Assets.php

```diff
@@ -96,7 +96,7 @@ class Assets
      * @var string string to identify Assetic filters using instanceof
      */
 
-    private $filterInterface = '\Assetic\Filter\FilterInterface';
+    private $filterInterface = '\\Assetic\\Filter\\FilterInterface';
```

Comment:
Le 24/09/2015 21:03, alain91 a écrit :
> Le 24/09/2015 19:02, Richard Griffith a écrit :
>> In htdocs/xoops_lib/Xoops/Core/Assets.php
>> <https://github.com/XOOPS/XoopsCore/pull/404#discussion_r40343942>:
>>
>>> @@ -96,7 +96,7 @@ class Assets
>>>       * @var string string to identify Assetic filters using instanceof
>>>       */
>>>
>>> -    private $filterInterface = '\Assetic\Filter\FilterInterface';
>>> +    private $filterInterface = '\\Assetic\\Filter\\FilterInterface';
>>
>> Why are we doubling up the backslashes? It is needed for double quoted
>> strings, but not in single quoted ones.
>>
>> A single quote needs to be escaped, so if there is a backslash at the
>> end of the string, that one needs to be escaped, i.e.
>> |'\Assetic\Filter\\' . $filtername|, but otherwise they are not
>> required. The '\' works, but make searching for string references to
>> classes much more complicated than it should be.
>>
>> Is there an issue I've missed that would make these required?
>>
>> —
>> Reply to this email directly or view it on GitHub
>> <https://github.com/XOOPS/XoopsCore/pull/404/files#r40343942>.
>>
>
>
> You are enterely right.
>
> I would double backslash to avoid confusing between sigle or double
> qouted strings. But I is not required.


but in php.net there is a strange explanation about backslah in single 
quoted strings:

Single quoted ¶

The simplest way to specify a string is to enclose it in single quotes 
(the character ').

To specify a literal single quote, escape it with a backslash (\). To 
specify a literal backslash, double it (\\). All other instances of 
backslash will be treated as a literal backslash: this means that the 
other escape sequences you might be used to, such as \r or \n, will be 
output literally as specified rather than having any special meaning.